overlays/nodejs/serverless/function.ts

```diff
@@ -160,18 +158,21 @@ export class Function extends pulumi.ComponentResource {
         let codePaths = computeCodePaths(
             closure, serializedFileNameNoExtension, options.includePaths, options.excludePackages);
 
-        // Create the Lambda Function.
-        this.lambda = new lambda.Function(name, {
+
+        // Copy over all option values into the function args.  Then overwrite anything we care
+        // about with our own values.  This ensures that clients can pass future supported
+        // lambda options without us having to know about it.
+        const copy = {
+            ...options,
             code: new pulumi.asset.AssetArchive(codePaths),
             handler: serializedFileNameNoExtension + "." + handlerName,
             runtime: options.runtime || lambda.NodeJS8d10Runtime,
-            environment: options.environment,
             role: this.role.arn,
             timeout: options.timeout === undefined ? 180 : options.timeout,
-            memorySize: options.memorySize,
-            deadLetterConfig: options.deadLetterConfig,
-            vpcConfig: options.vpcConfig,
```

Comment:
no need to pass along a random assortment of options.  We now copy everything over, but slam over all the properties this class is responsible for.